src/script/components/asset/AudioAsset.tsx

```diff
@@ -0,0 +1,163 @@
+import {formatSeconds} from 'Util/TimeUtil';
+
+import {AssetTransferState} from '../../assets/AssetTransferState';
+import type {ContentMessage} from '../../entity/message/ContentMessage';
+import type {FileAsset} from '../../entity/message/FileAsset';
+
+import RestrictedAudio from './RestrictedAudio';
+import SeekBar from './controls/SeekBar';
+import AudioSeekBar from './controls/AudioSeekBar';
+import AssetHeader from './AssetHeader';
+import MediaButton from './controls/MediaButton';
+
+import Icon from 'Components/Icon';
+import useEffectRef from 'Util/useEffectRef';
+import {useAssetTransfer} from './AbstractAssetTransferStateTracker';
+import {AssetRepository} from '../../assets/AssetRepository';
+import {registerReactComponent} from 'Util/ComponentUtil';
+
+const logger = getLogger('AudioAssetComponent');
+
+export interface AudioAssetProps {
+  assetRepository?: AssetRepository;
+  className?: string;
+  header: boolean;
+  // Does the asset have a visible header?
+  message: ContentMessage;
+  teamState?: TeamState;
+}
+
+const AudioAsset: React.FC<AudioAssetProps> = ({
+  header,
+  message,
+  className,
+  teamState = container.resolve(TeamState),
+  assetRepository = container.resolve(AssetRepository),
+}) => {
+  const asset = message.getFirstAsset() as FileAsset;
+  const [audioElement, setAudioElement] = useEffectRef<HTMLAudioElement>();
+  const isFileSharingReceivingEnabled = teamState.isFileSharingReceivingEnabled();
+  const {transferState, uploadProgress, cancelUpload} = useAssetTransfer(message);
+  const [audioTime, setAudioTime] = useState<number>(asset?.meta?.duration || 0);
+  const [audioSrc, setAudioSrc] = useState<string>();
+  const onTimeupdate = () => setAudioTime(audioElement.currentTime);
+  const showLoudnessPreview = !!(asset.meta?.loudness?.length > 0);
+  const onPauseButtonClicked = () => audioElement?.pause();
+
+  const onPlayButtonClicked = async () => {
+    if (audioSrc) {
+      audioElement?.play();
+    } else {
+      asset.status(AssetTransferState.DOWNLOADING);
+      try {
+        const blob = await assetRepository.load(asset.original_resource());
+        setAudioSrc(window.URL.createObjectURL(blob));
+        audioElement?.play();
+      } catch (error) {
+        logger.error('Failed to load audio asset ', error);
+      }
+      asset.status(AssetTransferState.UPLOADED);
+    }
+  };
+
+  useEffect(() => {
+    return () => {
+      window.URL.revokeObjectURL(audioSrc);
+    };
+  }, []);
+
+  return (
+    <div className={cx('audio-asset', className)} data-uie-name="audio-asset" data-uie-value={asset.file_name}>
+      <audio ref={setAudioElement} src={audioSrc} onTimeUpdate={onTimeupdate} />
+      {!message.isObfuscated() && (
+        <>
+          {header && (
+            <div style={{width: '100%'}}>
+              <AssetHeader message={message} />
+            </div>
+          )}
+          {!isFileSharingReceivingEnabled && <RestrictedAudio />}
```

Comment:
Could be a ternary operator

Reply:
https://github.com/wireapp/wire-webapp/pull/11577/commits/bee43b1c824461a571125a4848147818f55e919e